DentOS_Framework/DentOsTestbedLib/src/dent_os_testbed/lib/traffic/ixnetwork/ixnetwork_ixia_client_impl.py

```
@@ -663,3 +666,30 @@ def run_send_arp(self, device, command, *argv, **kwarg):
         return err, [{"success": msg["arg2"],
                       "port": msg["port"],
                       "src_ip": msg["src_ip"]} for msg in res]
+
+    def run_update_l1_config(self, device, command, *argv, **kwarg):
+        speeds_map = {"1000": "speed1000",
+                      "100": "speed100fd",
+                      "10000": "speed10g",
+                      "2500": "speed2.5g",
+                      "5000": "speed5g"}
+
+        if not IxnetworkIxiaClientImpl.ixnet:
+            return 0, "Ixia not connected"
+        if command == "update_l1_config":
+            vports = IxnetworkIxiaClientImpl.ixnet.Vport.find()
+            ports = kwarg['params'][0].get("ixia_ports", None)
+            speed = kwarg['params'][0].get("speed", None)
+            speed_to_ixia = None if speed is None else speeds_map[str(speed)]
```

`speed` and `ports` should be mandatory.
1. If the speed that the user passed did not match any speed in the `speeds_map` dict you should return an error.
2. At [L686 ](https://github.com/PLVision/dent-testing/pull/62/files#diff-14cdd5073bf03ba9e02b2ff30bd718aebe2db2d830a399d4093470bf4c8ceeccR686) if `ports` == `None` you will have an exception.

---

1. `speed ` can be `None` as for IXIA `update()` it is acceptable param.(in case we want to change `autoneg `only)
2. Agree on ports thing

---

you didn't resolve the issue with `ports` == `None`